src/components/views/donate/CryptoDonation.tsx

```diff
@@ -86,81 +90,379 @@ const CryptoDonation = (props: {
 	setSuccessDonation: SuccessFunction;
 	project: IProject;
 }) => {
-	const { setSuccessDonation } = props;
-	const { data: tokensList } = useQuery(FETCH_LISTED_TOKENS);
-
-	const [tokens, setTokensObject] = useState<ISelectObj[]>();
-	const [currentChainId, setCurrentChainId] = useState<number>(1);
-	const [selectedToken, setSelectedToken] = useState<ISelectObj[]>();
-
-	const buildTokensObj = (array: IToken[], chain: number) => {
-		const newArray = [tokensList];
-		array.forEach(e => {
-			if (e.chainId !== chain) return;
-			const obj: ISelectObj = {
-				label: e.symbol,
-				value: e.symbol,
-				chainId: e.chainId,
-				icon: `/images/tokens/${e.symbol?.toLocaleLowerCase()}.png`,
-			};
-			newArray.push(obj);
-		});
-		newArray.sort((a, b) => a.label.localeCompare(b.label));
-		// setSelectedToken(newArray && newArray[0])
-		return newArray;
-	};
+	const { chainId, account, library } = useWeb3React();
+	const {
+		state: { isEnabled, user, balance },
+		actions: { signIn },
+	} = UserContext();
+	const { project, setSuccessDonation } = props;
+	const { ethPrice } = usePrice();
+	const mainTokenPrice = new BigNumber(ethPrice).toNumber();
+	const networkId = chainId;
+	const [selectedToken, setSelectedToken] = useState<ISelectObj>();
+	const [selectedTokenBalance, setSelectedTokenBalance] = useState<any>();
+	const [customInput, setCustomInput] = useState<any>();
+	const [tokenPrice, setTokenPrice] = useState<any>(1);
+	const [amountTyped, setAmountTyped] = useState('');
+	const [geminiModal, setGeminiModal] = useState(false);
+	const [txHash, setTxHash] = useState<any>();
+	const [erc20List, setErc20List] = useState<any>();
+	const [erc20OriginalList, setErc20OriginalList] = useState<any>();
+	// TODO: Set this to a better flow, gotta discuss with design team but it is needed
+	const [unconfirmed, setUnconfirmed] = useState<any>();
+	const [inProgress, setInProgress] = useState<any>();
+	// const [anonymous, setAnonymous] = useState(false);
+	// const [selectLoading, setSelectLoading] = useState(false);
+	const [givBackEligible, setGivBackEligible] = useState(true);
+	const [showWalletModal, setShowWalletModal] = useState(false);
+	const [showDonateModal, setShowDonateModal] = useState(false);
+
+	const tokenSymbol = selectedToken?.symbol;
+	const isXdai = networkId === xdaiChain.id;
```

Comment:
It may have conflict with the logic on other places as I'm checking "xDAI" and the config file has "xDAI Chain". Would rather keep this file as I mentioned on the previous comment